impl/src/settings.rs

```diff
@@ -367,8 +380,95 @@ pub fn format_registry_url(registry_url: &str, name: &str, version: &str) -> Str
     .replace("{version}", version)
 }
 
+fn validate_crate_setting_additional_build_file_is_relative(
+  file_path: &Path,
+  crate_name: &str,
+  version: &VersionReq,
+) -> Result<(), RazeError> {
+  if file_path.is_absolute() {
+    return Err(RazeError::Config {
+      field_path_opt: Some(format!(
+        "raze.crates.{}.{}.additional_build_file",
+        crate_name,
+        version.to_string()
+      )),
+      message: "additional_build_file flags should be reletive paths from the manifest containing \
+                the definition"
+        .to_owned(),
+    });
+  }
+
+  Ok(())
+}
+
+/** Ensures crate settings which specify additional build files have canonicalized paths */
+fn validate_crate_setting_additional_build_file(
+  additional_build_file: &mut &mut PathBuf,
+  toml_path: &Path,
+  crate_name: &str,
+  version: &VersionReq,
+) -> Result<()> {
+  validate_crate_setting_additional_build_file_is_relative(
+    additional_build_file,
+    crate_name,
+    version,
+  )?;
+
+  **additional_build_file = toml_path.join(&additional_build_file).canonicalize()?;
+
+  Ok(())
+}
+
+/** Ensures crate settings associatd with the parsed [RazeSettings](crate::settings::RazeSettings) have valid crate settings */
+fn validate_crate_settings(settings: &mut RazeSettings, toml_path: &Path) -> Result<(), RazeError> {
```

Comment:
I saw this *after* seeing the above function, so I know what's going on. Just wanted to point out that readers inspecting this are going to be real suspicious that this function is doing something surreptitious seeing that it accepts the settings *mutably*. That's kind of the beauty of rust -- that it forces you to be up front when you're doing something weird like passing a mutable value to a "validation" function.

Reply:
What should I do about this? I updated the name to hopefully make this clearer but I'm open to alternative solutions.

Reply:
Fix the other issue, and this should no longer need to be `&mut RazeSettings`

Reply:
Continuing this here: https://github.com/google/cargo-raze/pull/301#discussion_r539913565

But this is also resolved.